WatchySim/WatchySim_SDL2.cpp

```diff
@@ -0,0 +1,428 @@
+        if (SDL_Init(SDL_INIT_VIDEO | SDL_INIT_TIMER) < 0) {
+            std::cerr << "SDL initialization failed: " << SDL_GetError() << std::endl;
+            return false;
+        }
+        
+        
+        // Create window with fixed size (same as Windows version)
+        window = SDL_CreateWindow("WatchySim",
+            SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
+            WINDOW_WIDTH, WINDOW_HEIGHT,
+            SDL_WINDOW_SHOWN);
+            
+        if (!window) {
+            std::cerr << "Window creation failed: " << SDL_GetError() << std::endl;
+            return false;
+        }
+        
+        
+        renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
+        if (!renderer) {
+            std::cerr << "Renderer creation failed: " << SDL_GetError() << std::endl;
+            return false;
+        }
+        
+        // Use nearest neighbor scaling for crisp pixels
+        SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "0");
+        
+        
+        // Create display texture
+        displayTexture = SDL_CreateTexture(renderer, SDL_PIXELFORMAT_ARGB8888,
+            SDL_TEXTUREACCESS_STREAMING, DISPLAY_WIDTH, DISPLAY_HEIGHT);
+            
+        if (!displayTexture) {
+            std::cerr << "Display texture creation failed: " << SDL_GetError() << std::endl;
+            return false;
+        }
+        
+        
+        // TODO: Load background image (for now, just use a gray background)
+        // In the Windows version, this loads from resources
+        // For Linux, we'll need to load from a file
+        
+        displayPixels = new uint32_t[DISPLAY_WIDTH * DISPLAY_HEIGHT];
+        clearDisplay();
+        
+        
+        // Set up timer for updates (1 second interval)
+        updateTimer = SDL_AddTimer(1000, timerCallback, this);
+        
+        
+        return true;
+    }
+    
+    void clearDisplay() {
+        memset(displayPixels, 0xFF, DISPLAY_WIDTH * DISPLAY_HEIGHT * sizeof(uint32_t));
+    }
+    
+    void drawBackground() {
+        // For now, just draw a gray background
+        SDL_SetRenderDrawColor(renderer, 128, 128, 128, 255);
+        SDL_RenderClear(renderer);
+        
+        // TODO: Draw the actual watch background image
+        // This would involve loading the background.png and rendering it
+    }
+    
+    void onPaint() {
+        // Clear the display buffer
+        clearDisplay();
+        
+        // Create SDL2 graphics context - use display dimensions, not window dimensions!
+        SDL2GraphicsContext context(renderer, displayTexture, displayPixels, 
+                                   DISPLAY_WIDTH, DISPLAY_HEIGHT);
+        
+        // Let Watchy draw the watch face
+        watchy.showWatchFace(&context);
+        
+        // Note: context.present() is called which updates the texture
+        // updateDisplay() will handle the actual rendering
+    }
+    
+    void drawTestPattern() {
```

Comment:
What is the goal of this? Would the default watch face not show that the infrastructure works?